src/metrics.c

```diff
@@ -213,7 +214,7 @@ static int metrics_set_gauge(metrics *m, char *name, double val, bool delta) {
  * @arg val The sample to add
  * @return 0 on success.
  */
-int metrics_add_sample(metrics *m, metric_type type, char *name, double val) {
+int metrics_add_sample(metrics *m, metric_type type, char *name, double val, double sample_rate) {
```

Comment:
@nspragg Yeah we can do it in a new PR, but for the increment counter as you said